R/key_colnames.R

```diff
@@ -1,19 +1,20 @@
 #' @export
 key_colnames.recipe <- function(x, ...) {
-  possible_keys <- c("geo_value", "time_value", "key")
-  keys <- x$var_info$variable[x$var_info$role %in% possible_keys]
-  keys[order(match(keys, possible_keys))] %||% character(0L)
+  geo_key <- x$var_info$variable[x$var_info$role %in% "geo_value"]
+  time_key <- x$var_info$variable[x$var_info$role %in% "time_value"]
+  keys <- x$var_info$variable[x$var_info$role %in% "key"]
+  c(geo_key, keys, time_key) %||% character(0L)
```

Comment:
It is preferrable for sliding: in the new PR we enforce that slides are applied per epi_key, so it makes sense to put time_value last.

Therefore we changed [arrange_canonical](https://github.com/cmu-delphi/epiprocess/blob/187bd5d87eea3a4acc323a715d1ba32edb34cd49/R/methods-epi_df.R#L424-L428) (after separating it into column and row arrangement because sometimes it's beneficial to apply just one). We also changed the order of the returned [key_colnames](https://github.com/cmu-delphi/epiprocess/blob/lcb/slide-improvements-2024-06/R/key_colnames.R#L24-L47) to support this.

Reply:
I should have been more clear. The second bullet just meant: "why not just reorder line 3 rather than changing all of it?"

Reply:
Ah I see... this is an instance of me rewriting a function in order to understand it 😅. I didn't quite follow how this code worked, especially `keys[order(match(keys, possible_keys))]`, but it makes a lot more sense now. I'd be happy to revert.